internal/cmd/stack/environment.go

```diff
@@ -73,6 +115,132 @@ func setVar(cliCtx *cli.Context) error {
+func (e *listEnvCommand) listEnv(cliCtx *cli.Context) error {
+	outputFormat, err := cmd.GetOutputFormat(cliCtx)
+	if err != nil {
+		return err
+	}
+
+	stackID := cliCtx.String(flagStackID.Name)
+
+	var query listEnvQuery
+	variables := map[string]interface{}{
+		"stack": graphql.ID(stackID),
+	}
+
+	if err := authenticated.Client.Query(cliCtx.Context, &query, variables); err != nil {
+		return err
+	}
+
+	var elements []listEnvElementOutput
+	for _, config := range query.Stack.RuntimeConfig {
+		var contextName *string
+		if config.Context != nil {
+			contextName = &config.Context.ContextName
+		}
+		if element, err := config.Element.toConfigElementOutput(contextName); err == nil {
+			elements = append(elements, element)
+		} else {
+			return err
+		}
+	}
+
+	switch outputFormat {
+	case cmd.OutputFormatTable:
+		return e.showOutputsTable(elements)
+	case cmd.OutputFormatJSON:
+		return e.showOutputsJSON(elements)
+	default:
+		return fmt.Errorf("unknown output format: %v", outputFormat)
+	}
+}
+
+func (e *listEnvCommand) showOutputsTable(outputs []listEnvElementOutput) error {
+	tableData := [][]string{{"Name", "Type", "Value", "File Mode", "Context"}}
+	for _, output := range outputs {
+		var row []string
+
+		row = append(row, output.Name)
+		row = append(row, output.Type)
+
+		var value string
+		switch {
+		case output.Runtime:
+			value = "<computed>"
+		case output.WriteOnly:
+			value = "*****"
+		case output.Type == string(fileTypeConfig):
+			value = output.trimmedValue()
+		case output.Value != nil:
+			value = *output.Value
+		default:
+			// keep value as empty string
+		}
+		row = append(row, value)
+
+		if output.FileMode != nil {
+			row = append(row, *output.FileMode)
+		} else {
+			row = append(row, "")
+		}
+		if output.Context != nil {
+			row = append(row, *output.Context)
+		} else {
+			row = append(row, "")
+		}
+
+		tableData = append(tableData, row)
+	}
+	return cmd.OutputTable(tableData, true)
+}
+
+func (e *listEnvCommand) showOutputsJSON(outputs []listEnvElementOutput) error {
+	return cmd.OutputJSON(outputs)
+}
+
+func (e *configElement) toConfigElementOutput(contextName *string) (listEnvElementOutput, error) {
+	var value = e.Value
+
+	if e.Type == fileTypeConfig && e.Value != nil {
+		if result, err := base64.StdEncoding.DecodeString(*e.Value); err != nil {
+			message := fmt.Sprintf("failed to decode base64-encoded file with id %s", e.ID)
+			return listEnvElementOutput{}, errors.Wrapf(err, message)
+		} else {
+			stringValue := string(result)
+			value = &stringValue
+		}
+	}
+
+	return listEnvElementOutput{
+		Name:     e.ID,
+		Type:     string(e.Type),
+		Value:    value,
+		FileMode: e.FileMode,
+		Context:  contextName,
+		Runtime:  e.Runtime,
+	}, nil
+}
+
+func (o *listEnvElementOutput) trimmedValue() string {
+	if o.Value == nil {
+		return ""
+	} else {
+		var result = *o.Value
+
+		valueNoCr := strings.ReplaceAll(*o.Value, "\r", "")
+		valueNoNewlines := strings.ReplaceAll(valueNoCr, "\n", " ")
```

Comment:
If the value isn't longer than the max length, the newlines aren't stripped correctly. Here's a test case highlighting it:

```go
{input: "abc\rdef\nghi\r\njkl", expected: "abc def ghi jkl"},
```

Reply:
Good catch, I've added your test case and adjusted the code.